python/tvm/topi/cuda/scatter.py

```diff
@@ -0,0 +1,278 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name, no-member, too-many-locals, too-many-arguments, too-many-statements, singleton-comparison, unused-argument
+"""Scatter operator """
+import tvm
+from tvm import te
+
+
+def gen_ir_1d(data, indices, updates, axis, out):
+    assert axis == 0
+    n = data.shape[0]
+
+    ib = tvm.tir.ir_builder.create()
+
+    out_ptr = ib.buffer_ptr(out)
+    data_ptr = ib.buffer_ptr(data)
+
+    bx = te.thread_axis("blockIdx.x")
```

Comment:
so we only use a single thread?

Reply:
I've spent most of today trying to get it to use multiple threads, but I'm running into an odd situation where the output gets padded with zeros if the output shape is larger than the input shape on any axis other than the updating axis. @tkonolige and I will keep looking at it to see if we can get a faster version to pass the tests.

Reply:
Great! Thanks.

Reply:
Thanks to help from @tkonolige and @tqchen, we figured out that we were accidentally launching one cuda kernel instead of 2. Fixed it and added threading to the PR.